### ISSUE TITLE
Exception error - 'Attempted to read or write protected memory' - VS2017 VB.net

### ISSUE THREAD

**Issue by Spiz103** [CoolProp/CoolProp#1623]
I have been attempting to integrate Coolprop's dll with a VB.net programme that I have been trying to write, however whenever I try to access the DLL the programme breaks and throws an error:

**System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'**

It has taken quite some time to get to this point, and I am pretty sure it is actually loading the DLL properly because I have had lots of other error messages at various points during development.

Machine is Windows 7 64-bit, with Coolprop 6.1.0 and 6.0.0 x64 seeming to make no difference.
Contrary to earlier reports I now have a functioning LibreOffice wrapper running the x64 DLL after numerous attempts.

Attached is my VB code

[Example.vb.txt](https://github.com/CoolProp/CoolProp/files/1572661/Example.vb.txt)

**Comment by jowr** [comment]
Thanks for creating the issue. Just to make sure: Why could not use the function prototypes from https://github.com/CoolProp/CoolProp/blob/master/wrappers/Excel/XLAM_module.bas ?

**Comment by jowr** [comment]
* You are missing the PtrSafe declaration
  ```vbnet
  Private Declare Function Props_private Lib "CoolProp.dll" Alias "PropsSI" (ByVal Output As String, ByVal Name1 As Integer, ByVal Value1 As Double, ByVal Name2 As Integer, ByVal Value2 As Double, ByVal Ref As String) As Double
  ```
  ```vbnet
  Private Declare PtrSafe Function PropsSI_private Lib "CoolProp.dll" Alias "PropsSI" (ByVal Output As String, ByVal Name1 As String, ByVal Value1 As Double, ByVal Name2 As String, ByVal Value2 As Double, ByVal Ref As String) As Double
  ```
* You might also want to use the error handling: 
  ```vbnet
  Private Function get_error_message() As String
    Dim errstring As String
    'Make a null-terminated string that is plenty big
    errstring = String(2000, vbNullChar)
    'Get the error string
    Call get_global_param_string_private("errstring", errstring, 2000)
    get_error_message = errstring
  End Function

  Public Function PropsSI(ByVal Output As String, ByVal Name1 As String, ByVal Value1 As Double, ByVal Name2 As String, ByVal Value2 As Double, ByVal fluid As String)
    On Error GoTo ErrorHandler
    Dim PropsSI_temp As Double

    PropsSI_temp = PropsSI_private(Output, Name1, Value1, Name2, Value2, fluid)
    
    If Abs(PropsSI_temp) > 1E+30 Then
        'Return error message
        PropsSI = get_error_message()
    Else
        PropsSI = PropsSI_temp
    End If
    Exit Function
    
  ErrorHandler:
    If Err = 13 Then
     Exit Function
    End If
    
    MsgBox "The most recent error number is " & Err & ". Its message text is: " & Error(Err)
    Exit Function
  End Function

**Comment by Spiz103** [comment]
Yes, I independently got it to work by reverse engineering the LibreOffice Wrapper back into VB.net

I must admit I was somewhat confused by the instructinos on the website referring to C# and all sorts which turned out to no longer be the case.
Regardless, thankyou for all the help - especially that code with the error handling, my reverse engineered one does not do that.

**Comment by jowr** [comment]
You are welcome, do you have any suggestions for how we can make things less confusing in the future? What was wrong in the documentation?

**Comment by Spiz103** [comment]
Currently the source archive on github for 6.1.0 gives a VB.net example which still contains the folders and such for the C# interface that used to be required.

Attached is the code I originally got working, and it might be best to simply provide a VB.net console programme consisting solely of that, or something similar to demonstrate the idea, along with a list of what functions can be exported and their parameters.

It would obviously require commenting, but I feel something like that is easier to intuit from.

Once again, thanks again.

[Simplexample.vb.txt](https://github.com/CoolProp/CoolProp/files/1575503/Simplexample.vb.txt)

**Comment by ibell** [comment]
That is not really a VB.net solution, it's more like an old-school VB
solution.  Nonetheless, thank you very much for that script.  Can you
manage a pull request on github?  If not, I can add that snippet to the
documentation.

**Comment by Spiz103** [comment]
Well my first programming language was QuickBASIC, which my father taught me when I was about 10-11, after which I went to VB6, then C++ and VB.net at University.
So unfortunately most of my programming is a port of QuickBasic in conception - but it seems to work!

Unfortunately I have no idea how to do anything on github, sorry. This is my literal first time on GitHub.

**Comment by henningjp** [comment]
@Spiz103 Take a look at the CoolProp Wiki.  I was in your shoes a year ago.  Also lots of help on Github help.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
